python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/sling/dagster_sling_translator.py

```diff
@@ -0,0 +1,196 @@
+        meta = config.get("meta", {})
+        deps = meta.get("dagster", {}).get("deps")
+        deps_out = []
+        if deps and isinstance(deps, str):
+            deps = [deps]
+        if deps:
+            assert isinstance(deps, list)
+            for asset_key in deps:
+                if cls.sanitize_stream_name(asset_key) != asset_key:
+                    raise ValueError(
+                        f"Deps Asset key {asset_key} for stream {stream_definition['name']} is not "
+                        "sanitized. Please use only alphanumeric characters and underscores."
+                    )
+                deps_out.append(AssetKey(asset_key.split(".")))
+            return deps_out
+
+        stream_name = stream_definition["name"]
+        components = cls.sanitize_stream_name(stream_name).split(".")
+        return [AssetKey(components)]
+
+    @classmethod
+    @public
+    def get_description(cls, stream_definition: Mapping[str, Any]) -> Optional[str]:
+        config = stream_definition.get("config", {}) or {}
+        if "sql" in config:
+            return config["sql"]
+        meta = config.get("meta", {})
+        description = meta.get("dagster", {}).get("description")
+        return description
+
+    @classmethod
+    @public
+    def get_metadata(cls, stream_definition: Mapping[str, Any]) -> JsonMetadataValue:
+        return MetadataValue.json(stream_definition.get("config", {}))
+
+    @classmethod
+    @public
+    def get_group_name(cls, stream_definition: Mapping[str, Any]) -> Optional[str]:
+        config = stream_definition.get("config", {}) or {}
+        meta = config.get("meta", {})
+        return meta.get("dagster", {}).get("group")
+
+    @classmethod
+    @public
+    def get_freshness_policy(
+        cls, stream_definition: Mapping[str, Any]
+    ) -> Optional[FreshnessPolicy]:
+        config = stream_definition.get("config", {}) or {}
+        meta = config.get("meta", {})
+        freshness_policy_config = meta.get("dagster", {}).get("freshness_policy")
+        if freshness_policy_config:
+            return FreshnessPolicy(
+                maximum_lag_minutes=float(freshness_policy_config["maximum_lag_minutes"]),
+                cron_schedule=freshness_policy_config.get("cron_schedule"),
+                cron_schedule_timezone=freshness_policy_config.get("cron_schedule_timezone"),
+            )
+
+    @classmethod
+    @public
+    def get_auto_materialize_policy(
+        cls, dbt_resource_props: Mapping[str, Any]
+    ) -> Optional[AutoMaterializePolicy]:
+        config = dbt_resource_props.get("config", {}) or {}
+        meta = config.get("meta", {})
+        auto_materialize_policy_config = meta.get("dagster", {}).get("auto_materialize_policy")
+        if auto_materialize_policy_config:
+            if auto_materialize_policy_config.get("type") == "eager":
+                return AutoMaterializePolicy.eager()
+            elif auto_materialize_policy_config.get("type") == "lazy":
```

Comment:
I believe lazy AMPs are slowly getting pushed out. You might be able to cut this entire branch operation out and assume eager.

Also, how would you configure this AMP?

Reply:
Yea they are but thought I'd leave it in there since they might still exist in the wild. 
I have a unit test for the config, but haven't tested the actual implementation:

```
                "config": {"meta": {"dagster": {"auto_materialize_policy": {"type": "eager"}}}},
```